cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -71,4 +76,12 @@ public void run() {
     LOG.info("File cleanup completed, Successful file deletes - {}. Failed file deletes - {}. Log Cleanup took {} ms",
              deleteCount, failureCount, (completionTime - startTime));
   }
+
+  private void deleteParentFoldersIfEmpty(String logFilePath) {
+    File folder = new File(logFilePath).getParentFile();
```

Comment:
Why it is local file only?? Log files are stored using `Location`, hence all operations should be using `Location`, not `File`.

Reply:
Replaced File classes with Location.